src/toast/pipeline_tools/classes.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2019 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+import os
+import pickle
+import sys
+
+import numpy as np
+
+from ..timing import function_timer, Timer
+from ..tod import AnalyticNoise
+from ..utils import Logger, Environment
+from .. import qarray
+
+
+def name2id(name, maxval=2 ** 16):
+    """ Map a name into an index.
+    """
+    value = 0
+    for c in name:
+        value += ord(c)
+    return value % maxval
+
+
+class Focalplane:
+    detector_data = {}
+    _detweights = None
+    sample_rate = None
+    _radius = None
+    _detquats = None
+    _noise = None
+
+    def __init__(
+        self, detector_data=None, fname_pickle=None, sample_rate=None, radius_deg=None
+    ):
+        """ Instantiate a focalplane
+
+        Args:
+            fname_pickle (str) :  Pickle file containing the focal
+                 plane dictionary
+            sample_rate (float) :  Default sampling rate for all
+                detectors.  Will be overridden by 'fsample' fields
+                if they exist for the detectors in the dictionary.
+            radius_deg (float) :  force the radius of the focal plane.
+                otherwise it will be calculated from the detector
+                offsets.
+        """
+        if detector_data is not None:
+            self.detector_data.update(detector_data)
+        if fname_pickle is not None:
+            with open(fname_pickle, "rb") as picklefile:
+                self.detector_data.update(pickle.load(picklefile))
+        self.sample_rate = sample_rate
+        self._radius = radius_deg
+        return
+
+    def refresh(self):
```

Comment:
what about `reset_properties` instead of `refresh`?

Reply:
Works for me.